### PR TITLE
clientv3: filter learner members during autosync

### PR DESCRIPTION
**Pull request by chrisayoub**
This change is to ensure that all members returned during the client's
AutoSync are started and are not learners, which are not valid
etcd members to make requests to.

I have tested this patch with Kubernetes as a client and it works as intended.

### CODE REVIEW

File: client/v3/client.go
Comment (ahrtr) [inline client/v3/client.go:191]: Can you add an unit test?

Comment (chrisayoub) [reply client/v3/client.go:191]: @ahrtr just added!

File: client/v3/client_test.go
Comment (ahrtr) [inline client/v3/client_test.go:267]: Suggest to add a field something like  `members []*etcdserverpb.Member` into `mockCluster`, and update MemberList to return this `members` field instead of a hard code list.

Comment (chrisayoub) [reply client/v3/client_test.go]: Done, good suggestion, thanks

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you.

cc @serathius @ptabor

**Comment by chrisayoub** [comment]
@ahrtr it says `First-time contributors need a maintainer to approve running workflows.` I assume I need some sort of approval for this to merge?

**Review by serathius — APPROVED** [review]

**Comment by chrisayoub** [comment]
@serathius anything else we need here before merging?

**Comment by serathius** [comment]
Second maintainer approval. cc @ptabor @spzala

**Review by spzala — APPROVED** [review]
Thanks @chrisayoub 
lgtm

**Comment by serathius** [comment]
Note, I think we should have an e2e tests with learners.

**Pull request merged**
